autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -185,31 +186,41 @@ def update_execution(status: ExecutionStatus) -> ExecutionResult:
             ):
                 yield execution
 
-        # Release lock on credentials ASAP
-        if creds_lock:
-            creds_lock.release()
-
-        r = update_execution(ExecutionStatus.COMPLETED)
-        s = input_size + output_size
-        t = (
-            (r.end_time - r.start_time).total_seconds()
-            if r.end_time and r.start_time
-            else 0
-        )
-        db_client.spend_credits(user_id, credit, node_block.id, input_data, s, t)
-
     except Exception as e:
+        end_status = ExecutionStatus.FAILED
```

Comment:
we should also release here tbh if speed is the goal

Reply:
the main code is not releasing before `upsert_execution_output` too, I don't want to introduce any side-effects.